lib/mocha/class_method.rb

```diff
@@ -66,11 +63,8 @@ def remove_new_method
 
     def restore_original_method
       return if use_prepended_module_for_stub_method?
-      if stub_method_overwrites_original_method?
-        original_method_owner.send(:define_method, method_name, original_method_body)
-      end
-      return unless original_visibility
-      Module.instance_method(original_visibility).bind(original_method_owner).call(method_name)
+      original_method_owner.send(:define_method, method_name, original_method_body) if stub_method_overwrites_original?
```

Comment:
Again, I'm not very keen on these long lines, especially when the condition is on the end of the line, because I think it's easy to miss. I should really setup a Rubocop rule to enforce this!

Also it would be nice to keep stylistic changes like this separate from the core refactoring here.